### PR TITLE
feat: Java support

### PR DESCRIPTION
**Pull request by campionfellin**
*Issue #, if available:*

https://github.com/eladb/jsii-srcmak/issues/1

*Description of changes:*

* Updates README with contributing tips
* Adds java support to this package

TODO: the java that's output into the snapshots looks good to me, but I want to `yarn link` and test with `cdk8s` first.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by campionfellin** [comment]
@eladb snapshot tests will continue to fail due to generated date in the 

`@javax.annotation.Generated(value = \\"jsii-pacmak/1.5.0 (build 46538f8)\\", date = \\"2020-05-27T04:36:59.445Z\\")` annotation.

Is it worth it to add a flag to `jsii-pacmak` like `--no-annotations` or `--no-date-annotation`?

File: README.md [hunk 1]
```diff
@@ -67,7 +67,33 @@ The output directory will include a python module that corresponds to the
...
+$ jsii-srcmak /src/dir --java-outdir=dir --java-package=hello.world
+```
+
+* The `outdir`/`--java-outdir` option points to the root directory of your Java project.
+* The `package`/`--java-package` option is the java package name.
+
+The output directory will include a java module that corresponds to the
+original module. This code depends on the following maven package (should be defined directly or indirectly in the project's `pom.xml` file):
+
+- [jsii](https://mvnrepository.com/artifact/software.amazon.jsii)
```
Comment (eladb) [inline README.md:96]: Also worth mentioning that the output will include a tarball resource that must be bundled in the project (maybe provide an example snippet from a `pom.xml`?)

File: lib/options.ts [hunk 2]
```diff
@@ -23,6 +23,12 @@ export interface Options {
    * @default - python is not generated
    */
   python?: PythonOutputOptions;
+
+  /**
+   * Produce java code.
+   * @default - java is not generated
```
Comment (eladb) [inline lib/options.ts:29]: Add a note here that says that the code will be produced under `src` (with an example of the full path combined with the package name). Also mention the requirements from `Pom.xml` (dependency on jsii runtime and resource).

File: lib/srcmak.ts [hunk 3]
```diff
@@ -33,5 +33,12 @@ export async function srcmak(srcdir: string, options: Options = { }) {
       const target = path.join(options.python.outdir, reldir);
       await fs.move(source, target, { overwrite: true });
     }
+    
+    if (options.java) {
+      const reldir = options.java.package.replace(/\./g, '/');
+      const source = path.resolve(path.join(workdir, 'dist/java/src/main/java', reldir));
```
Comment (eladb) [inline lib/srcmak.ts:39]: Why is it `src/main`? Is this where things are expected conventionally by the Pom.xml?

File: test/util.ts [hunk 4]
```diff
@@ -1,30 +1,40 @@
 import * as fs from 'fs-extra';
 import * as path from 'path';
 
+interface excludeOptions {
```
Comment (eladb) [inline test/util.ts:4]:
```suggestion
interface SnapshotOptions {
```

File: test/util.ts [hunk 5]
```diff
@@ -1,30 +1,40 @@
...
+  excludeLines?: RegExp[];
+  excludeFiles?: string[];
+}
+
 /**
  * Returns a dictionary where keys are relative file names and values are the
  * file contents. Useful to perform snapshot testing against full directories.
  */
-export async function snapshotDirectory(basedir: string, exclude: string[] = [], reldir = '.'): Promise<Record<string, string>> {
+export async function snapshotDirectory(basedir: string, excludeOptions: excludeOptions = {}, reldir = '.'): Promise<Record<string, string>> {
```
Comment (eladb) [inline test/util.ts:13]:
```suggestion
export async function snapshotDirectory(basedir: string, options: SnapshotOptions = {}, reldir = '.'): Promise<Record<string, string>> {
```

**Pull request merged**